src/payment/strategies/afterpay/afterpay-payment-strategy.ts

```diff
@@ -101,28 +101,23 @@ export default class AfterpayPaymentStrategy implements PaymentStrategy {
         return this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
     }
 
-    private _redirectToAfterpay(countryName: string, paymentMethod?: PaymentMethod): void {
+    private _redirectToAfterpay(countryCode: string, paymentMethod?: PaymentMethod): void {
         if (!this._afterpaySdk || !paymentMethod || !paymentMethod.clientToken) {
             throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
         }
 
-        this._afterpaySdk.initialize({ countryCode: this._mapCountryToISO2(countryName)});
+        this._afterpaySdk.initialize({ countryCode });
         this._afterpaySdk.redirect({ token: paymentMethod.clientToken });
     }
 
-    private _mapCountryToISO2(countryName: string): string {
-        switch (countryName) {
-        case 'Australia':
-            return 'AU';
-
-        case 'New Zealand':
-            return 'NZ';
-
-        case 'United States':
-            return 'US';
+    private _mapCurrencyToISO2(currencyCode: string): string {
+        const countryByCurrency: { [key: string]: string } = {
+            AUD: 'AU',
+            NZD: 'NZ',
+            CAD: 'CA',
+            USD: 'US',
+        };
 
-        default:
-            return 'AU';
-        }
+        return countryByCurrency[currencyCode];
```

Comment:
Can we add a check here in the case if currencyCode passed here is `''` or something else other than the specified countries?